src/utils/validateOptions.js

```diff
@@ -0,0 +1,51 @@
+export default function ({ result, ruleName, possible, actual }) {
+
+  if (Array.isArray(possible)) {
+    if (isValid(possible, actual)) { return }
+    result.warn({
```

Comment:
When I try this PR on my local test-bed, the reporter outputs `[object Object] [stylelint]` instead of the message. Is this because you're passing just an object into `result.warn()`, rather than a message string and object?

Is the API changing in `4.2` and you're working against that?